File: app/Modules/Attendance/Models/AttendanceCorrectionRequest.php
<?php

declare(strict_types=1);

namespace App\Modules\Attendance\Models;

use App\Models\User;
use App\Modules\Attendance\Models\AttendanceLog;
use App\Modules\Employees\Models\Employee;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class AttendanceCorrectionRequest extends Model
{
    protected $table = 'attendance_correction_requests';

    protected $fillable = [
        'attendance_log_id',
        'user_id',
        'employee_id',
        'proposed_check_in_time',
        'proposed_check_out_time',
        'status',
        'reason',
        'reviewed_by',
        'reviewed_at',
        'review_comment',
    ];

    protected $casts = [
        'attendance_log_id' => 'int',
        'user_id' => 'int',
        'employee_id' => 'int',
        'proposed_check_in_time' => 'datetime',
        'proposed_check_out_time' => 'datetime',
        'reviewed_by' => 'int',
        'reviewed_at' => 'datetime',
        'created_at' => 'datetime',
        'updated_at' => 'datetime',
    ];

    public function attendanceLog(): BelongsTo
    {
        return $this->belongsTo(AttendanceLog::class, 'attendance_log_id');
    }

    public function user(): BelongsTo
    {
        return $this->belongsTo(User::class, 'user_id');
    }

    public function employee(): BelongsTo
    {
        return $this->belongsTo(Employee::class, 'employee_id');
    }

    public function reviewer(): BelongsTo
    {
        return $this->belongsTo(User::class, 'reviewed_by');
    }
}

File: mobile/lib/models/attendance_correction_request.dart
class AttendanceCorrectionRequest {
  final int id;
  final int attendanceLogId;
  final int userId;
  final int? employeeId;
  final DateTime? proposedCheckInTime;
  final DateTime? proposedCheckOutTime;
  final String status;
  final String reason;
  final int? reviewedBy;
  final DateTime? reviewedAt;
  final String? reviewComment;

  AttendanceCorrectionRequest({
    required this.id,
    required this.attendanceLogId,
    required this.userId,
    required this.employeeId,
    required this.proposedCheckInTime,
    required this.proposedCheckOutTime,
    required this.status,
    required this.reason,
    required this.reviewedBy,
    required this.reviewedAt,
    required this.reviewComment,
  });

  factory AttendanceCorrectionRequest.fromJson(Map<String, dynamic> json) {
    DateTime? parseDt(dynamic v) {
      if (v == null) return null;
      if (v is String) return DateTime.tryParse(v);
      return null;
    }

    return AttendanceCorrectionRequest(
      id: (json['id'] as num?)?.toInt() ?? 0,
      attendanceLogId: (json['attendance_log_id'] as num?)?.toInt() ?? 0,
      userId: (json['user_id'] as num?)?.toInt() ?? 0,
      employeeId: (json['employee_id'] as num?)?.toInt(),
      proposedCheckInTime: parseDt(json['proposed_check_in_time']),
      proposedCheckOutTime: parseDt(json['proposed_check_out_time']),
      status: json['status'] as String? ?? 'pending',
      reason: json['reason'] as String? ?? '',
      reviewedBy: (json['reviewed_by'] as num?)?.toInt(),
      reviewedAt: parseDt(json['reviewed_at']),
      reviewComment: json['review_comment'] as String?,
    );
  }

  String get statusLabel {
    switch (status) {
      case 'pending':
        return 'Pending';
      case 'approved':
        return 'Approved';
      case 'rejected':
        return 'Rejected';
      default:
        return status;
    }
  }
}
